ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -17,6 +17,122 @@
 {-# LANGUAGE TupleSections              #-}
 {-# LANGUAGE TypeApplications           #-}
 
+{-|
+  = Snapshot managing and contents
+
+  Snapshotting a ledger state means saving a copy of the in-memory part of the
+  ledger state serialized as a file on disk, as well as flushing differences
+  between the last snapshotted ledger state and the one that we are snapshotting
```

Comment:
The snapshot also contains a copy (aka "restore point" aka `tables` below) of the matching on-disk state. Just flushing is not enough, we need to a make a copy that won't be altered by subsequent rights.